opcua/common/event.py

```diff
@@ -22,53 +22,140 @@ class Event(object):
+        if message:
+            self.Message = ua.LocalizedText(message)
+        elif not self.event.Message:
+            self.event.Message = ua.LocalizedText(self.source.get_browse_name().Text)
+        self.isession.subscription_service.trigger_event(self.event)
+
+
+def get_event_from_node(node):
+    if node.nodeid.Identifier in ua.uaevents_auto.IMPLEMNTED_EVENTS.keys():
+        return ua.uaevents_auto.IMPLEMNTED_EVENTS[node.nodeid.Identifier]()
+    else:
+        parent_identifier, parent_eventtype = _find_parent_eventtype(node)
+        if not parent_eventtype:
+            return None
+
+        class CustomEvent(parent_eventtype):
+
+            def __init__(self):
+                super(CustomEvent, self).__init__()
+                curr_node = node
+                while curr_node.nodeid.Identifier != parent_identifier:
+                    properties = curr_node.get_referenced_nodes(refs=ua.ObjectIds.HasProperty, direction=ua.BrowseDirection.Forward)
+                    for prop in properties:
+                        setattr(self, prop.get_browse_name().Name, prop.get_value())
+
+                    parents = node.get_referenced_nodes(refs=ua.ObjectIds.HasSubtype, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+                    if len(parents) != 1: # Something went wrong
+                        return None
+                    curr_node = parents[0]
+
+            #TODO: Extend to show all fields of CustomEvent
+            def __str__(self):
+                s = 'CustomEvent('
+                s += 'EventId:{}'.format(self.EventId)
+                s += ', EventType:{}'.format(self.EventType)
+                s += ', SourceNode:{}'.format(self.SourceNode)
+                s += ', SourceName:{}'.format(self.SourceName)
+                s += ', Time:{}'.format(self.Time)
+                s += ', RecieveTime:{}'.format(self.RecieveTime)
+                s += ', LocalTime:{}'.format(self.LocalTime)
+                s += ', Message:{}'.format(self.Message)
+                s += ', Severity:{}'.format(self.Severity)
+                s += ')'
+
+        #class CustomEvent():
+
+            #pass
+    #references = node.get_children_descriptions(refs=ua.ObjectIds.HasProperty)
+    #for desc in references:
+        #child = Node(self.isession, desc.NodeId)
+        #setattr(self.event, desc.BrowseName.Name, child.get_value())
+
+    return CustomEvent()
+
+
+def _find_parent_eventtype(node):
+    parents = node.get_referenced_nodes(refs=ua.ObjectIds.HasSubtype, direction=ua.BrowseDirection.Inverse, includesubtypes=False)
+
+    if len(parents) != 1:   # Something went wrong
+        return None
+    if parents[0].nodeid.Identifier in ua.uaevents_auto.IMPLEMNTED_EVENTS.keys():
+        return node.nodeid.Identifier, ua.uaevents_auto.IMPLEMNTED_EVENTS[parents[0].nodeid.Identifier]
+    else:
+        _find_parent_eventtype(parents[0])
+
+
+
+
+class CustomEvent(ua.BaseEvent):
```

Comment:
Uhhh, the second should not exist any more.... This is not completely cleaned version... I will clean it with other stuff, sorry :/